core/src/cost_update_service.rs

```diff
@@ -174,6 +190,36 @@ impl CostUpdateService {
         );
         dirty
     }
+
+    // Based historical data, it seems reasonable to flag it as large change if updated_cost
+    // is more than double of input_data;
+    // Compare updated_cost against input_cost instead of previous calculated cost captures
+    // how far apart between single data point and calculated value, which is a good indicator
+    // of potential issues.
```

Comment:
nit: "A drastic difference between the updated cost and the cost prior to the update is a good indicator of potential issues."

This comment seems more helpful as a code comment inside `update_cost_model()` where this function is called